libraries/libstratosphere/source/settings/impl/settings_key_value_store.cpp

```diff
@@ -0,0 +1,1629 @@
+        }
+
+        Result ReadSystemDataFirmwareDebug(u64 *out_count, char * const out_buffer, size_t out_buffer_size) {
+            /* Check preconditions. */
+            AMS_ASSERT(out_count != nullptr);
+            AMS_ASSERT(out_buffer);
+
+            /* Get the firmware debug system data. */
+            SystemData *system_data = nullptr;
+            R_TRY(GetSystemData<SystemDataTag::Fwdbg>(std::addressof(system_data), ncm::SystemDataId::FirmwareDebugSettings));
+            AMS_ASSERT(system_data != nullptr);
+
+            /* Read the data. */
+            return ReadAllBytes<SystemData>(system_data, out_count, out_buffer, out_buffer_size);
+        }
+
+        Result ReadSystemDataPlatformConfiguration(u64 *out_count, char * const out_buffer, size_t out_buffer_size) {
+            /* Check preconditions. */
+            AMS_ASSERT(out_count != nullptr);
+            AMS_ASSERT(out_buffer);
+
+            ncm::SystemDataId system_data_id;
+
+            switch (GetSplHardwareType()) {
+                case SplHardwareType_None:
+                    *out_count = 0;
+                    return ResultSuccess();
+                case SplHardwareType_Icosa:
+                    system_data_id = ncm::SystemDataId::PlatformConfigIcosa;
+                    break;
+                case SplHardwareType_IcosaMariko:
+                    system_data_id = ncm::SystemDataId::PlatformConfigIcosaMariko;
+                    break;
+                case SplHardwareType_Copper:
+                    system_data_id = ncm::SystemDataId::PlatformConfigCopper;
+                    break;
+                case SplHardwareType_Hoag:
+                    system_data_id = ncm::SystemDataId::PlatformConfigHoag;
+                    break;
+                case SplHardwareType_Calcio:
+                    system_data_id = ncm::SystemDataId::PlatformConfigCalcio;
+                    break;
+                case SplHardwareType_Aula:
+                    system_data_id = ncm::SystemDataId::PlatformConfigAula;
+                    break;
+                AMS_UNREACHABLE_DEFAULT_CASE();
+            }
+
+            /* Get the platform configuration system data. */
+            SystemData *system_data = nullptr;
+            R_TRY(GetSystemData<SystemDataTag::PfCfg>(std::addressof(system_data), system_data_id));
+            AMS_ASSERT(system_data != nullptr);
+
+            /* Read the data. */
+            return ReadAllBytes<SystemData>(system_data, out_count, out_buffer, out_buffer_size);
+        }
+
+        Result ReadSystemSaveData(u64 *out_count, char * const out_buffer, size_t out_buffer_size) {
+            /* Check preconditions. */
+            AMS_ASSERT(out_count != nullptr);
+            AMS_ASSERT(out_buffer);
+        
+            /* Get the system save data. */
+            SystemSaveData *system_save_data = nullptr;
+            if (const auto result = GetSystemSaveData(std::addressof(system_save_data), false); R_FAILED(result)) {
```

Comment:
mfw you did it correctly here

Use the same style as I pasted for the previous two here.